src/Validation/URLValidationRESTController.php

```diff
@@ -207,16 +253,45 @@ public function validate_post_url( $request ) {
 			);
 		}
 
-		$validity = $this->url_validation_provider->get_url_validation( $url, get_post_type( $post_id ) );
+		$post_type = get_post_type( $post_id );
+		$data      = $this->get_validation_data( $url, $post_type );
+
+		return rest_ensure_response( $this->filter_response_by_context( $data, $request['context'] ) );
+	}
+
+	/**
+	 * Validate page by URL.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function validate_url( $request ) {
+		$data = $this->get_validation_data( $request['url'] );
+
+		return rest_ensure_response( $this->filter_response_by_context( $data, $request['context'] ) );
+	}
```

Comment:
Based on what I've outlined above, there would not be a need for a `validate-url` endpoint. All that is needed is a way to obtain the nonce to pass as the `amp_validate` query parameter. This wouldn't necessarily need to be exposed in the REST API either. It could be passed directly to the client as part of the page exports from PHP.

Reply:
This has been [reverted](https://github.com/ampproject/amp-wp/pull/6610#:~:text=%E2%80%A6-,b43afa2,-Merge%20state) in favor of the approach proposed in https://github.com/ampproject/amp-wp/pull/6610#discussion_r711431170